src/indexer/allo/v1/roleRevoked.ts

```diff
@@ -0,0 +1,71 @@
+import { EventHandlerArgs } from "chainsauce";
+import type { Indexer } from "../../indexer.js";
+import { Changeset } from "../../../database/index.js";
+import { parseAddress } from "../../../address.js";
+import { PROGRAM_ADMIN_ROLE } from "./roles.js";
+
+export default async function handleEvent(
+  args: EventHandlerArgs<
+    Indexer,
+    "AlloV1/ProgramFactory/V1" | "AlloV2/Registry/V1",
+    "RoleRevoked"
+  >
+): Promise<Changeset[]> {
+  const {
+    chainId,
+    event,
+    context: { db, logger },
+  } = args;
+
+  switch (args.event.contractName) {
```

Comment:
maybe this could be an early return to reduce number of blocks here? does this make sense:

```
if(args.event.contractName !== "AlloV1/ProgramFactory/V1") {
   return [];
}

// event handler here
```

Reply:
@boudra I'm going to add more contracts now so the switch should make more sense